doc/classes/Thread.xml

```diff
@@ -60,6 +66,26 @@
 				Returns [constant OK] on success, or [constant ERR_CANT_CREATE] on failure.
 			</description>
 		</method>
+		<method name="suspend" qualifiers="static">
+			<return type="Variant" />
+			<param index="0" name="resume_signal" type="Signal" />
+			<param index="1" name="connect_on_main_thread" type="bool" default="true" />
+			<description>
+				Suspend a thread until the signal provided in the [param resume_signal] argument is emitted. As connecting signals between threads is often discouraged (or blocked by thread guards), the [param connect_on_main_thread] is provided to perform the connection to the signal on the main thread.
+				Similarly to script 'await', if the signal provided is emitted with a single argument, then this function will return it. For multiple arguments, an array is returned.
+				As an additional note, if you intend to await on a script function instead of a signal, use the following syntax instead: [code] Thread.suspend( function().completed ) [/code].
+				This function will fail if called from the main thread.
+
+				Code example using this function with GDScript 'await'.
```

Comment:
I dont think this header is needed, in html docs user has a picker to select what version of the code to look at (if available) and in editor i beliece gdscript version is always shown

![image](https://github.com/user-attachments/assets/3b07d59b-98c0-43e3-8174-f4b18e7a8a9e)